### PR TITLE
LTP: fixed fchmod03&04 and fchown03

### PR DESCRIPTION
**Pull request by vamsikrishna935**
Tests were modified to create temporary directories using root permissions.

### CODE REVIEW

File: tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
++ to create directories using nobody.
+diff --git a/testcases/kernel/syscalls/fchmod/fchmod03.c b/testcases/kernel/syscalls/fchmod/fchmod03.c
+index a5824f36b..7330e6234 100644
+--- a/testcases/kernel/syscalls/fchmod/fchmod03.c
++++ b/testcases/kernel/syscalls/fchmod/fchmod03.c
+@@ -141,7 +141,7 @@ void setup(void)
+        tst_sig(NOFORK, DEF_HANDLER, cleanup);
+
+        tst_require_root();
+-
```
Comment (shaikshavali1) [inline tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: Please keep blank line.

File: tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch [hunk 2]
```diff
@@ -0,0 +1,90 @@
++ Created and deleted temporary directories with root permissions as tests were failing
++ to create directories using nobody.
```
Comment (shaikshavali1) [inline tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: Please add the links to the github issues 236 and 357.  If possible include test case names (fchmod03, fchmod04 & fchown03) in patch description.

File: tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+         * Create a test file under temporary directory with specified
+@@ -165,6 +165,7 @@ void cleanup(void)
+        if (close(fd) == -1)
+                tst_resm(TWARN | TERRNO, "close failed");
+
++       tst_require_root(); // TODO: Will be removed once fixing git issue :357
+        tst_rmdir();
+
+ }
+diff --git a/testcases/kernel/syscalls/fchmod/fchmod04.c b/testcases/kernel/syscalls/fchmod/fchmod04.c
```
Comment (shaikshavali1) [inline tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: Please use complete test case name in patch file like... "fix_fchmod03_fchmod04_fchown03.patch". or you can use separate .patch file for each test case.

File: tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+index befaadb35..823ab0870 100644
+--- a/testcases/kernel/syscalls/fchmod/fchmod04.c
++++ b/testcases/kernel/syscalls/fchmod/fchmod04.c
+@@ -145,7 +145,7 @@ int main(int ac, char **av)
+ void setup(void)
+ {
+        tst_require_root();
+-
```
Comment (shaikshavali1) [inline tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: Please take care of formatting.

File: tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        tst_rmdir();
+diff --git a/testcases/kernel/syscalls/fchown/fchown03.c b/testcases/kernel/syscalls/fchown/fchown03.c
+index f7fe994fd..d7e276460 100644
+--- a/testcases/kernel/syscalls/fchown/fchown03.c
++++ b/testcases/kernel/syscalls/fchown/fchown03.c
+@@ -111,13 +111,13 @@ static void setup(void)
+        tst_sig(FORK, DEF_HANDLER, cleanup);
+
+        tst_require_root();
+-
```
Comment (shaikshavali1) [inline tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: Please take care of formatting.

File: tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+-       tst_tmpdir();
++//     tst_tmpdir(); // TODO: Will be enabled once fixing git issue :236
+
+        fildes = SAFE_OPEN(cleanup, TESTFILE, O_RDWR | O_CREAT, FILE_MODE);
+
+@@ -134,6 +134,6 @@ static void cleanup(void)
+ {
+        if (fildes > 0 && close(fildes))
+                tst_resm(TWARN | TERRNO, "close(%s) Failed", TESTFILE);
+-
```
Comment (shaikshavali1) [inline tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: Please take care of formatting.

File: tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch (hunk 4 above)
Comment (vamsikrishna935) [reply tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: @shaikshavali1 , Could you please elaborate what formatting needs to be done, i am not clear. Here indentation looks good, what else ?

Comment (shaikshavali1) [reply tests/ltp/patches/ltp_fchmod03_04_fchown03_fix.patch]: keep the blank line.

**Comment by vamsikrishna935** [comment]
[](https://github.com/lsds/sgx-lkl/issues/236)
[](https://github.com/lsds/sgx-lkl/issues/357)

**Review by hukoyu — APPROVED** [review]
LGTM

**Pull request merged**